palace/models/waveportoperator.cpp

```diff
@@ -599,12 +599,13 @@ WavePortData::WavePortData(const config::WavePortData &data,
   //            given frequency, Math. Comput. (2003).
   // See also: Halla and Monk, On the analysis of waveguide modes in an electromagnetic
   //           transmission line, arXiv:2302.11994 (2023).
-  mu_eps_min = 0.0;  // Use standard inverse transformation to avoid conditioning issues
-                     // associated with shift
-  // const double c_max = mat_op.GetLightSpeedMax().Max();
-  // MFEM_VERIFY(c_max > 0.0 && c_max < mfem::infinity(),
-  //             "Invalid material speed of light detected in WavePortOperator!");
-  // mu_eps_min = 1.0 / (c_max * c_max);
+  const double c_max = mat_op.GetLightSpeedMax().Max();
+  MFEM_VERIFY(c_max > 0.0 && c_max < mfem::infinity(),
+              "Invalid material speed of light detected in WavePortOperator!");
+  mu_eps_min = 1.0 / (c_max * c_max) * 0.5;  // Add a safety factor for minimum propagation
+                                             // constant possible
```

Comment:
What's the origin of this safety factor? Is it a floating point issue?

Reply:
Yeah, if the wave port is specified on a waveguide supporting a TEM mode, then setting the safety factor of 1 will lead to the shift being directly on top of the actual eigenvalue which is going to  cause problems. Setting the shift to 0 works fine (and was what we did previously) but having some non-zero shift can speed up the eigenvalue solve convergence. It didn't make a significant difference in my testing but I figured it might as well be some positive value.